frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Outlet } from 'react-router-dom';
+
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useEffect, useNavigate } from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+import { type UserGetAllItemResponseDto } from '~/packages/users/users.js';
+
+const ProtectedRoute = (): React.ReactElement => {
+  const navigate = useNavigate();
+  const { users } = useSelector(
+    (user: { users: UserGetAllItemResponseDto[]; dataStatus: string }) => user,
+  );
+  const hasUser = Boolean(users);
+
+  useEffect(() => {
+    storage
+      .get(StorageKey.TOKEN)
+      .then((token) => {
+        if (!token && !hasUser) {
+          navigate(AppRoute.SIGN_IN);
+        }
+      })
+      .catch((error) => {
+        const typedError = error as Error;
+        throw new Error(typedError.message);
+      });
+  }, [navigate, hasUser]);
```

Comment:
```suggestion
  if (!hasUser) {
  // redirect
```

Reply:
So, everythig that i did is a base of previose expirience. Token should be saved in localstorage and  it should compare with some instanse  in repository and so on.... But here is an other aprouch and I thing it more progressive. So i fixed acording  rules and hints and demands. So Please check my work again. I' am a little bit confused that my code is too short now compare with previouse ones